toqito/rand/random_circulant_gram.py

```diff
@@ -0,0 +1,38 @@
+"""Generate random circulant Gram matrix."""
+import numpy as np
+
+
+def random_circulant_gram(dim: int) -> np.ndarray:
+    r"""Generate a random circulant Gram matrix of specified dimension.
+
+    This function creates a circulant matrix based on a random diagonal matrix and the normalized Discrete
```

Comment:
It might be better to add some details/ref on what a circulant matrix is in this docstring + more details on what a normalized DFT is. 

https://en.wikipedia.org/wiki/Circulant_matrix
https://ccrma.stanford.edu/~jos/st/Normalized_DFT.html

Reply:
I have made the changes, please review it and let me know if further modification is required.

Reply:
Might want to double-check how references are added. Other docstrings might be helpful for you.

Reply:
@ankit-pn I think you'll want to add the links you provided within the references in `docs/articles.bib`. So for instance,

```bib
@misc{WikiCirculantMat,
  author  = "Wikipedia",
  title   = "Circulant matrix",
  howpublished = {https://en.wikipedia.org/wiki/Circulant_matrix}
  
}
```

And then use the `:cite:` command in the docstring (as done elsewhere in `toqito`) instead of putting the whole link there. Similarly for your other link as well.